library/core/src/main/java/com/google/android/exoplayer2/text/cea/CeaDecoder.java

```diff
@@ -97,6 +97,8 @@ public SubtitleOutputBuffer dequeueOutputBuffer() throws SubtitleDecoderExceptio
     if (availableOutputBuffers.isEmpty()) {
       return null;
     }
+    // check if 608 decoder needs to clean up the stale caption
+    clearStuckCaptions();
     // iterate through all available input buffers whose timestamps are less than or equal
```

Comment:
Comment from our internal review, about where to put the check function ('clearStuckCaptions()`):

>  @bjitivo(Bryan Ischo @ TiVo Corp.) 22 hours ago
> I don't know this code well but the way I read it, the only time we're going to check on whether or not to clear stuck captions is when a non-empty output buffer has arrived. But wouldn't that clear captions anyway?
> 
> And if cc are stuck on the screen and no non-empty output buffers are not coming, then this code path will not be entered and the cc will not be cleared.
> 
> Why aren't you just resetting a 16 second timer whenever cc is displayed and then clearing captions when the 16 second timer goes off (which implies no captioning commands for 16 seconds)?

>   @stevemayhew
> stevemayhew Pending
> Using timers in a handler is problematic,  but it is pretty unclear that dequeueOutputBuffer() is called every call to TextRenderer.render(). The render() method is called every 16ms for all ExoPlayer renderers, so it is probably clearer to either put it in setPosition() or even better add an abstract method to the SubtitleDecoder (checkStuckCaptions() with a default that does nothing and override that, call it from the render() method in TextRenderer

Reply:
again, `dequeueOutputBuffer()` is the best, unless we want to add methods to `CeaDecoder`